### ISSUE TITLE
dbAppendTable fails from R

### ISSUE THREAD

**Issue by noamross** [duckdb/duckdb#1593]
Running `DBI::dbAppend()` consistently fails with a DuckDB backend from R, as below. I am running the current GitHub master, and get the same error on both Linux and MacOS.

``` r
library(duckdb)
#> Loading required package: DBI
library(DBI)

# Make a data frame with a list-column
z <- data.frame(a = 1:10, b = letters[1:10])
conn <- dbConnect(duckdb())
dbWriteTable(conn, "test", z)
dbAppendTable(conn, "test", z)
#> Error in dbBind(res, params): duckdb_bind_R: bind parameter values need to have length 1
```

<sup>Created on 2021-04-11 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 4.0.4 (2021-02-15)
#>  os       Ubuntu 20.04.2 LTS          
#>  system   x86_64, linux-gnu           
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       America/New_York            
#>  date     2021-04-11                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version    date       lib source                      
#>  assertthat    0.2.1      2019-03-21 [2] RSPM (R 4.0.3)              
#>  backports     1.2.1      2020-12-09 [1] RSPM (R 4.0.3)              
#>  cli           2.3.1      2021-02-23 [1] RSPM (R 4.0.3)              
#>  conflicted    1.0.4      2019-06-21 [1] RSPM (R 4.0.0)              
#>  crayon        1.3.4.9000 2020-08-20 [1] local (r-lib/crayon@4bceba8)
#>  DBI         * 1.1.1      2021-01-15 [1] RSPM (R 4.0.3)              
#>  debugme       1.1.0      2017-10-22 [1] RSPM (R 4.0.0)              
#>  digest        0.6.27     2020-10-24 [1] RSPM (R 4.0.3)              
#>  dotenv        1.0.2      2017-03-01 [1] RSPM (R 4.0.0)              
#>  duckdb      * 0.2.6      2021-04-11 [1] local                       
#>  ellipsis      0.3.1      2020-05-15 [2] RSPM (R 4.0.3)              
#>  evaluate      0.14       2019-05-28 [2] RSPM (R 4.0.3)              
#>  fansi         0.4.2      2021-01-15 [1] RSPM (R 4.0.3)              
#>  fs            1.5.0      2020-07-31 [1] RSPM (R 4.0.2)              
#>  glue          1.4.2      2020-08-27 [1] RSPM (R 4.0.2)              
#>  highr         0.8        2019-03-20 [2] RSPM (R 4.0.3)              
#>  htmltools     0.5.1.1    2021-01-22 [1] RSPM (R 4.0.3)              
#>  knitr         1.31       2021-01-27 [1] RSPM (R 4.0.3)              
#>  lifecycle     1.0.0      2021-02-15 [1] RSPM (R 4.0.3)              
#>  magrittr      2.0.1      2020-11-17 [1] RSPM (R 4.0.3)              
#>  pillar        1.5.1      2021-03-05 [1] RSPM (R 4.0.3)              
#>  pkgconfig     2.0.3      2019-09-22 [2] RSPM (R 4.0.3)              
#>  purrr         0.3.4      2020-04-17 [2] RSPM (R 4.0.3)              
#>  reprex        1.0.0      2021-01-27 [1] RSPM (R 4.0.3)              
#>  RhpcBLASctl   0.20-137   2020-05-17 [1] RSPM (R 4.0.0)              
#>  rlang         0.4.10     2020-12-30 [1] RSPM (R 4.0.3)              
#>  rmarkdown     2.7        2021-02-19 [1] RSPM (R 4.0.3)              
#>  rstudioapi    0.13       2020-11-12 [1] RSPM (R 4.0.3)              
#>  sessioninfo   1.1.1      2018-11-05 [2] RSPM (R 4.0.3)              
#>  stringi       1.5.3      2020-09-09 [1] RSPM (R 4.0.2)              
#>  stringr       1.4.0      2019-02-10 [2] RSPM (R 4.0.3)              
#>  styler        1.4.1      2021-03-30 [1] RSPM (R 4.0.4)              
#>  tibble        3.1.0      2021-02-25 [1] RSPM (R 4.0.3)              
#>  utf8          1.1.4      2018-05-24 [1] RSPM (R 4.0.0)              
#>  vctrs         0.3.7      2021-03-29 [1] RSPM (R 4.0.4)              
#>  withr         2.4.1      2021-01-26 [1] RSPM (R 4.0.3)              
#>  xfun          0.22       2021-03-11 [1] RSPM (R 4.0.3)              
#>  yaml          2.2.1      2020-02-01 [2] RSPM (R 4.0.3)              
#> 
#> [1] /home/noamross/R/x86_64-pc-linux-gnu-library/4.0
#> [2] /usr/local/lib/R/site-library
#> [3] /usr/local/lib/R/library
```

</details>

**Comment by hannesmuehleisen** [comment]
This is also broken in the release version, but I agree, it should be fixed. Thanks for reporting!

**Comment by hannesmuehleisen** [comment]
Ah, the issue is that `dbAppendTable` is a fairly new generic in DBI and we did not have an implementation yet. What works now is `dbWriteTable(con, "test", z, append=TRUE)` .

**Comment by hannesmuehleisen** [comment]
This should be fixed now in #2134 thanks to @krlmlr

**Comment by hannesmuehleisen** [comment]
I tried your example on the latest master, it works now.
